File: guess_number/guess_number.cc
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main() {
    srand(time(0));
    auto number = (rand() % 5)+1;

    int guess;
    auto guesses = 1;

    printf("guess 1 - 5: ");
    if ( !scanf("%d", &guess)) guess = -1;

    for ( ; guess != number; guesses++) {
        if( guess < number)
            puts("guess higher!");
        else
            puts("guess lower!");
        if ( !scanf("%d", &guess)) guess = -1;
    }

    printf("you got it in %d tries\n", guesses);
}

File: guess_number/guess_number.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func main() {
	rand.Seed(time.Now().UnixNano())
	number := rand.Intn(5) + 1

	var guess int
	guesses := 1

	fmt.Printf("guess 1-5: ")
	_, e := fmt.Scanf("%d", &guess)
	if e != nil {
		guess = -1
	}

	for ; guess != number; guesses += 1 {
		if guess < number {
			fmt.Println("guess higher!")
		} else {
			fmt.Println("guess lower!")
		}

		_, e = fmt.Scanf("%d", &guess)

		if e != nil {
			guess = -1
		}
	}

	fmt.Printf("you got it in %d tries\n", guesses)
}
